Fix CONC-396, "!includedir in my.cnf does not include all .cnf and .ini files in the directory"

actapia opened the pull request:
These changes fix the issue [CONC-396](https://jira.mariadb.org/browse/CONC-396). Previously, `_mariadb_read_options` and `_mariadb_read_options_from_file` would only read options from files named `my.cnf` within the specified directory when the `!includedir` directive was encountered. With these changes, the functions find all `.cnf` (and, on Windows, `.ini`) files in the included directory.

9EOR9 commented:
Thanks for your PR . Would you mind to squash your commits into a single commit?

actapia committed:
Fix CONC-396.
!includedir now includes all .cnf or .ini files, sorted alphabetically,
in the specified directory.

actapia commented:
Okay, I've squashed the commits.

9EOR9 committed:
Merge branch '3.4' into conc-396

9EOR9 committed:
Update connection.c

Fixed previous commit
